lib/simple_forms_api_submission/s3.rb

```diff
@@ -14,13 +14,22 @@ def initialize(region:, access_key_id:, secret_access_key:, bucket_name:)
       @bucket_name = bucket_name
     end
 
-    def upload_file(key, file)
-      obj = resource.bucket(bucket_name).object(key)
-      obj.upload_file(file)
+    def put_object(key, file, metadata = {})
+      Datadog::Tracing.trace('S3 Put File(s)') do
+        # Convert nil values to empty strings in the metadata
+        metadata&.transform_values! { |value| value || '' }
 
-      { success: true }
-    rescue => e
-      { success: false, error_message: "S3 Upload failure for #{file}: #{e.message}" }
+        client.put_object({
+                            bucket: Settings.ivc_forms.s3.bucket,
+                            key:,
+                            body: File.read(file),
```

Comment:
I think the File.open comment is referring to this line here

Reply:
Yup I changed it from a read to an open